src/main/java/com/redhat/devtools/lsp4ij/settings/ui/LanguageServerPanel.java

```diff
@@ -241,4 +256,25 @@ public ComboBox<ErrorReportingKind> getErrorReportingKindCombo() {
         return errorReportingKindCombo;
     }
 
+    public @NotNull List<ValidationInfo> doValidateAll() {
+        List<ValidationInfo> validations = new ArrayList<>();
+        var serverNameWidget = getServerName();
+        if (serverNameWidget != null) {
+            serverNameWidget.validate(validations);
+        }
+        var commandLineWidget = getCommandLine();
+        if (commandLineWidget != null) {
+            commandLineWidget.validate(validations);
+        }
+        return validations;
+    }
+
+    private void addValidator(JTextComponent textComponent) {
+        textComponent.getDocument().addDocumentListener(new DocumentAdapter() {

```

Comment:
Don't do that here but in the ValidableConsoleWidget impl in the `initializeValidator(ValidatableDialog dialog)` method

Reply:
I think addValidator should just add in a list CommandLineWidget, ServerNameWidget (list of ValidatableConsoleWidget) and in doValidateAll should just iterate from this list.

```java
private void addValidatableWidget(@Nullable ValidatableConsoleWidget validatableWidget) {
   if (validatableWidget == null) {
      return;
   }
   validatableWidget.initializeValidator(this);
   this.validatableWidgets.add(validatableWidget)
}
```